processing/src/main/java/org/apache/druid/query/aggregation/SingleValueAggregator.java

```diff
@@ -0,0 +1,111 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.druid.query.aggregation;
+
+import org.apache.druid.error.DruidException;
+import org.apache.druid.error.InvalidInput;
+import org.apache.druid.segment.ColumnValueSelector;
+
+import javax.annotation.Nullable;
+
+
+public class SingleValueAggregator implements Aggregator
+{
+  final ColumnValueSelector selector;
+
+  @Nullable
+  Object value;
+
+  private boolean isNullResult = true;
+
+  private boolean isAggregateInvoked = false;
+
+  public SingleValueAggregator(ColumnValueSelector selector)
+  {
+    this.selector = selector;
+  }
+
+  @Override
+  public void aggregate()
+  {
+    if (isAggregateInvoked) {
+      throw InvalidInput.exception("Subquery expression returned more than one row");
+    }
+    boolean isNotNull = (selector.getObject() != null);
+    if (isNotNull) {
+      isNullResult = false;
+      value = selector.getObject();
```

Comment:
Should probably reuse the `.getObject()` called above, because for primitives that would cause autoboxing.

I was going to suggest using `.isNull()` check at the top instead, however, I don't think they might behave correctly for object selectors, therefore probably refraining from the suggestion.